src/clib/pio_spmd.c

```diff
@@ -291,7 +302,7 @@ int pio_swapm(void *sendbuf, int *sendcounts, int *sdispls, MPI_Datatype *sendty
              * used to choose between mpi_irsends and mpi_isends - the default
              * is still mpi_irsend
              */
-            if (isend)
+            if (handshake && isend)
```

Comment:
I don't think we can guarantee that a recv is posted without an application handshake (since we post recvs in blocks).